pkg/proxy/proxy.go

```diff
@@ -185,32 +203,77 @@ func (p *Proxy) forwardRequest(host string, w http.ResponseWriter, req *http.Req
 		// If the client sent multiple XFF headers, flatten them.
 		proxyReq.Header.Set(xff, strings.Join(previous, ", "))
 	}
-	proxyReq.URL.Host = host
 	proxyReq.URL.Scheme = "https"
+	attempts := 0
 
-	log.Debug("Forwarding request to %s", proxyReq.URL.String())
-	client := http.Client{}
-	resp, err := client.Do(proxyReq)
-	if err != nil {
-		if urlErr, ok := err.(*url.Error); ok && urlErr.Timeout() {
-			writeJSONError(w, http.StatusGatewayTimeout, urlErr)
-		} else {
+	for {
+		proxyReq.URL.Host = acct.Host
+		log.Debug("Forwarding request to %s", proxyReq.URL.String())
+		client := http.Client{}
+		result, err := client.Do(proxyReq)
+
+		if err != nil {
+			if urlErr, ok := err.(*url.Error); ok && urlErr.Timeout() {
+				writeJSONError(w, http.StatusGatewayTimeout, urlErr)
+			} else {
+				writeJSONError(w, http.StatusBadGateway, err)
+			}
+			result.Body.Close()
```

Comment:
Might `result` be nil at this point? Must we close anything if there’s an error?

Reply:
from the doc of `client.Do`, `On error, any Response can be ignored. A non-nil Response with a non-nil error only occurs when CheckRedirect fails, and even then the returned Response.Body is already closed.` I can remove the close call.